src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,158 @@
+        chekIndex(index, size - 1);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        chekIndex(index, size - 1);
+        Node<T> node = getNode(index);
+        T oldNodeValue = node.value;
+        node.value = value;
+        return oldNodeValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        chekIndex(index, size - 1);
+        return unlink(getNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> node = headElement;
+        for (int i = 0; i < size; i++) {
+            if (node.value != null && node.value.equals(object) || node.value == object) {
+                unlink(node);
+                return true;
+            }
+            node = node.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void chekIndex(int index, int size) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+    }
+
+    private void setTailElement(T element) {
+        final Node<T> lastNode = tailElement;
+        final Node<T> newNode = new Node<>(lastNode, element, null);
+        tailElement = newNode;
+        if (lastNode == null) {
+            headElement = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> neededNode;
+        if (index < size / 2) {
+            neededNode = headElement;
+            for (int i = 0; i < index; i++) {
+                neededNode = neededNode.next;
+            }
+        } else {
+            neededNode = tailElement;
+            for (int i = size - 1; i > index; i--) {
+                neededNode = neededNode.prev;
+            }
+        }
+        return neededNode;
+    }
+
+    private void getElementBefore(T value, Node<T> element) {
+        Node<T> prevElement = element.prev;
+        Node<T> newElement = new Node<>(prevElement, value, element);
+        element.prev = newElement;
+        if (prevElement == null) {
+            headElement = newElement;
+        } else {
+            prevElement.next = newElement;
+        }
+        size++;
+    }
+
+    private T unlink(Node<T> unlinkNode) {
+        Node<T> prev = unlinkNode.prev;
+        Node<T> next = unlinkNode.next;
+
+        if (prev == null) {
+            headElement = next;
+        } else {
+            prev.next = next;
+            unlinkNode.prev = null;
+        }
+
+        if (next == null) {
+            tailElement = prev;
+        } else {
+            next.prev = prev;
+            unlinkNode.next = null;
+        }
+
+        T value = unlinkNode.value;
+        unlinkNode.value = null;
```

Comment:
Same here, it's redundant
no need to do unlinkNode.value = null;
Garbage Collector will take care of it